internal/venice-client-common/src/main/java/com/linkedin/venice/compute/ComputeRequestWrapper.java

```diff
@@ -153,4 +156,20 @@ public void setOperations(List<ComputeOperation> operations) {
         throw new VeniceException("Compute request version " + version + " is not support yet.");
     }
   }
+
+  public void initializeOperationResultFields(Schema resultSchema) {
+    List<ComputeOperation> operations = getOperations();
+    this.operationResultFields = new ArrayList<>(operations.size());
+    ComputeOperation computeOperation;
+    ReadComputeOperator operator;
+    for (int i = 0; i < operations.size(); i++) {
+      computeOperation = operations.get(i);
+      operator = ComputeOperationType.valueOf(computeOperation).getOperator();
+      this.operationResultFields.add(resultSchema.getField(operator.getResultFieldName(computeOperation)));
+    }
+  }
+
+  public List<Schema.Field> getOperationResultFields() {
```

Comment:
Shall we combine `getOperationResultFields` and `initializeOperationResultFields`?
For example, we can keep a reference of `resultSchema` in `ComputeRequestWrapper` and when calling function: `getOperationResultFields`, it can check whether the cached `operationResultFields` is null or not, and if it is `null`, do the initialization.

With two functions, we need to make sure the caller will call `initializeOperationResultFields` before `getOperationResultFields`.

Reply:
If I always had the result schema at construction-time, then I could pre-init at that time, but the code using this class does not make it very easy, IIRC. I'll look into it again and see if I can improve this.

Reply:
FYI... I've been working on this this morning, and I'm finding a LOT of cruft. There is definitely room to simplify many things and make the code less error-prone, but it's going to be a big refactoring. Not sure that it should be clubbed into this change. I can keep going for now, but LMK if you prefer that we just go ahead with the smaller change, since it seems kind of critical. I could always do the bigger refactoring afterwards...